packages/react/docs/guides/onboarding.md

```diff
@@ -0,0 +1,259 @@
+# Onboarding
+
+The purpose of this document is to help new developers quickly and easily onboard to this project and
+pick up the styles and practices we use to make this a coherent code base.
+
+## Coding practices
+
+Below you will find major items we strive to provide in all of our code and try to check for in all
+code reviews.
+
+### Prop naming conventions
+
+When naming props all names should be in camelCase, events should be prefixed with `on` (onClick), and booleans should be prefixed with `is` or `has` (isActive, hasLabel).
+
+### Localization
+
+All non-falsy translatable strings should be passed in an `i18n` prop and the defaultProps should include
```

Comment:
Should we say something about variables and substring substitution? Not sure if we have a standard there yet, but at least I think we can agree that we are moving away from using functions as i18n values.

For graphite they wanted to use indexes rather than variable names right?

Reply:
Good call. I've added the example we discussed.